gui/src/hooks/firmware-update.ts

```diff
@@ -26,47 +24,40 @@ const hash = (str: string) => {
   return (hash >>> 0) / 2 ** 32;
 };
 
-const uniqueUserKey = `${await hostname()}-${await locale()}-${platform()}-${version()}`;
-
 const firstAsset = (assets: any[], name: string) =>
   assets.find((asset: any) => asset.name === name && asset.browser_download_url);
 
-const processDeployData = (sortedMap: [string, string][]) => {
-  const deployData: DeployData = new Map();
-  let minTime = 0;
-  for (const [percent, date] of sortedMap) {
-    const d = new Date(date);
-    if (d.getTime() <= minTime) return null; // Dates in the wrong order / cancel
-    minTime = d.getTime();
-    deployData.set(parseFloat(percent), new Date(date));
-  }
-  return deployData;
-};
-
-const checkUserCanUpdate = async (deployAssetUrl: string) => {
-  if (!deployAssetUrl) return false;
-  const deployDataJson: DeployData | null = await fetch(deployAssetUrl)
+const checkUserCanUpdate = async (url: string) => {
+  if (!url) return true;
+  const deployDataJson = await fetch(url)
```

Comment:
not gonna cache the request?

Reply:
There is almost no rate limit on raw.github. Should be fine